lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -38,23 +40,52 @@ def get_version
     def bulk(actions)
       @action_count ||= 0
       @action_count += actions.size
-      
+
       return if actions.empty?
-      bulk_body = actions.collect do |action, args, source|
+
+
+      bulk_actions = actions.collect do |action, args, source|
         args, source = update_action_builder(args, source) if action == 'update'
 
         if source && action != 'delete'
           next [ { action => args }, source ]
         else
           next { action => args }
         end
-      end.
-      flatten.
-      reduce("") do |acc,line|
-        acc << LogStash::Json.dump(line)
-        acc << "\n"
       end
 
+      bulk_body = ""
+      bulk_responses = []
+      bulk_actions.each do |action|
+        as_json = action.is_a?(Array) ?
```

Comment:
`update_action_builder` returns either a a single action or an array of them (some bulk things take up two lines, like indexing). Before this was solved via a call to `.flatten`